### PR TITLE
[WIP] Misc OSP updates

### PR DESCRIPTION
**Pull request by shawndwells**
work in progress

### CODE REVIEW

File: linux_os/guide/services/openstack/keystone/keystone_lockout_failure_attempts.var [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+title: 'Maximum Number of Failed Authentication Attempts'
+
+description: |-
+    Specifies the maximum number of failed authentication attempts allowed
+    prior to disabling a Keystone user.
+
+type: string
+
+operator: less than or equal
```
Comment (shawndwells) [inline linux_os/guide/services/openstack/keystone/keystone_lockout_failure_attempts.var]: Is this a correct var operator? Where do I find docs?

Comment (yuumasato) [reply linux_os/guide/services/openstack/keystone/keystone_lockout_failure_attempts.var]: You may be mistaking XCCDF for OVAL. The operator for the variable should be in test, no?

**Comment by shawndwells** [comment]
On 6/4/18 10:54 AM, Watson Yuuma Sato wrote:
>
> In 
> linux_os/guide/services/openstack/keystone/keystone_lockout_failure_attempts.var 
> <https://github.com/OpenSCAP/scap-security-guide/pull/2894#discussion_r192770046>:
>
> > @@ -0,0 +1,18 @@
> +documentation_complete: true
> +
> +title: 'Maximum Number of Failed Authentication Attempts'
> +
> +description: |-
> +    Specifies the maximum number of failed authentication attempts allowed
> +    prior to disabling a Keystone user.
> +
> +type: string
> +
> +operator: less than or equal
>
> You may be mistaking XCCDF for OVAL. The operator for the variable 
> should be in test, no?
>

No idea. Not sure what this 'operator' field is used for and doesn't 
appear documented anywhere.

Using the templates of all the other variable files:

> $ cd linux_os/ && grep -rin operator
>
> guide/system/selinux/selinux-booleans/var_samba_create_home_dirs.var:12:operator: 
> equals
> guide/system/selinux/selinux-booleans/var_samba_domain_controller.var:12:operator: 
> equals
> guide/system/selinux/selinux-booleans/var_samba_enable_home_dirs.var:12:operator: 
> equals
> guide/system/selinux/selinux-booleans/var_samba_export_all_ro.var:12:operator: 
> equals
> guide/system/selinux/selinux-booleans/var_samba_export_all_rw.var:12:operator: 
> equals
> guide/system/selinux/selinux-booleans/var_samba_load_libgfapi.var:12:operator: 
> equals
> guide/system/selinux/selinux-booleans/var_samba_portmapper.var:12:operator: 
> equals
> guide/system/selinux/selinux-booleans/var_samba_run_unconfined.var:12:operator: 
> equals
> guide/system/selinux/selinux-booleans/var_samba_share_fusefs.var:12:operator: 
> equals
> guide/system/selinux/selinux-booleans/var_samba_share_nfs.var:12:operator: 
> equals

**Comment by mpreisler** [comment]
we don't parse the `operator` key in `.var` files, I think it's there by mistake, I am not sure why it's there TBH

**Comment by redhatrises** [comment]
> we don't parse the `operator` key in `.var` files, I think it's there by mistake, I am not sure why it's there TBH

That must be a bug, because `operator` should be processed in `.var` files as it is part of the XCCDF spec and has the following valid options:
```
- equals
- not equal
- greater than
- less than
- greater than or equal
- less than or equal
- pattern match
```

File: linux_os/guide/services/openstack/keystone/keystone_lockout_failure_attempts.var (hunk 1 above)
Comment (mpreisler) [reply linux_os/guide/services/openstack/keystone/keystone_lockout_failure_attempts.var]: I have confirmed that this is an issue in the yaml parser, I am working on a fix.

Comment (mpreisler) [reply linux_os/guide/services/openstack/keystone/keystone_lockout_failure_attempts.var]: https://github.com/OpenSCAP/scap-security-guide/pull/2900 fixes this, review would be great

Comment (mpreisler) [reply linux_os/guide/services/openstack/keystone/keystone_lockout_failure_attempts.var]: #2900 has been merged, this needs a rebase and then the operator will work. it uses the same values XCCDF allows.

**Comment by mpreisler** [comment]
> we don't parse the operator key in .var files, I think it's there by mistake, I am not sure why it's there TBH

I have fixed the situation and now it's parsed and supported. "interactive" is also now parsed and supported.

**Comment by shawndwells** [comment]
On 6/4/18 4:07 PM, Martin Preisler wrote:
> I have fixed the situation and now it's parsed and supported. 
> "interactive" is also now parsed and supported.

Thanks @mpreisler! Will work on this profile a bit more tomorrow afternoon.

**Comment by dahaic** [comment]
Hey @shawndwells I know you have lot on your plate. Can you outline what needs to be done for you to be satisfied with the PR? Maybe someone can take it and finish it. To clean up your PR queue. :)

**Comment by dahaic** [comment]
@openscap-ci , test this please.

**Comment by redhatrises** [comment]
Closing because branch no longer exists.

**Pull request closed without merging**